backend/src/main/java/com/tradomator/service/WalletService.java

```diff
@@ -0,0 +1,40 @@
+package com.tradomator.service;
+
+import com.tradomator.model.Wallet;
+import com.tradomator.repo.WalletRepo;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+
+@Service
+public class WalletService {
+
+    private final WalletRepo walletRepo;
+
+
+    public WalletService(WalletRepo walletRepo) {
+        this.walletRepo = walletRepo;
+
+    }
+
+
+    public Wallet getLatestWalletDocument() {
+        List<Wallet> all = walletRepo.findAll();
+
+        if(all.isEmpty()){
+            throw new NoSuchElementException("No wallets in DB found!");
+        }
+
+        Wallet latestWallet = null;
+        long max = 0;
+        for ( Wallet wallet : all){
+            if( wallet.getUpdateTime() > max ){
+                max = wallet.getUpdateTime();
+                latestWallet = wallet;
+            }
+        }
+        return latestWallet;
```

Comment:
Maybe create a repo function for this, should look something like this
```suggestion
    return walletRepo.findFirstByOrderByUpdatedTimeDesc()
    .orElseThrow(throw new NoSuchElementException("No wallets in DB found!"));
```

Reply:
elegant: tried to implement like this. did't work at first. With a little adaption finally worked :   
`return walletRepo.findTopByOrderByUpdateTimeDesc().orElseThrow( () -> new NoSuchElementException("Wallet Id is not set or no wallets in DB found!"));`